### PR TITLE
Fix valgrind issues

### PR DESCRIPTION
**Pull request by rafiw**
this is based on #252 and #414

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1427/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1428/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1429/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1430/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1431/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1432/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1434/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1437/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1442/ for details (Mellanox internal link).

**Comment by rafiw** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1445/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1447/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1452/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1454/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1455/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1456/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1457/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1460/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1461/ for details (Mellanox internal link).

File: contrib/jenkins_tests/vg.sh [hunk 1]
```diff
@@ -4,77 +4,112 @@ source $(dirname $0)/globals.sh
...
 do_check_filter "Checking for valgrind ..." "on"
 
-do_module "tools/valgrind"
+do_module "tools/valgrind-3.12.0"
 
 cd $WORKSPACE
 
 rm -rf $vg_dir
 mkdir -p $vg_dir
 cd $vg_dir
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/vg.sh]: please add `set +eE`

File: contrib/jenkins_tests/vg.sh [hunk 2]
```diff
@@ -4,77 +4,112 @@ source $(dirname $0)/globals.sh
...
 
-${WORKSPACE}/configure --prefix=${vg_dir} --with-valgrind $jenkins_test_custom_configure
+${WORKSPACE}/configure --prefix=${vg_dir}/install --with-valgrind $jenkins_test_custom_configure
 
 make $make_opt all
+make install
 rc=$?
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/vg.sh]: please add `set -eE`

**Comment by DanielLibenson** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1472/ for details (Mellanox internal link).

File: contrib/test_jenkins.sh [hunk 3]
```diff
@@ -137,6 +137,16 @@ for target_v in "${target_list[@]}"; do
...
+	if [ 8 -lt "$jenkins_opt_exit" -o "$rc" -eq 0 ]; then
+	    if [ "$jenkins_test_vg" = "yes" ]; then
+	        $WORKSPACE/contrib/jenkins_tests/vg.sh
+	        ret=$?
+	        if [ $ret -gt 0 ]; then
+	           do_err "case: [vg: rc=$rc]"
+	        fi
+	        rc=$((rc + $ret))
+	    fi
+    fi
```
Comment (igor-ivanov) [inline contrib/test_jenkins.sh]: @rafiw do not forget move this back

File: contrib/jenkins_tests/vg.sh (hunk 2 above)
Comment (rafiw) [inline contrib/jenkins_tests/vg.sh]: done

File: contrib/jenkins_tests/vg.sh (hunk 1 above)
Comment (rafiw) [inline contrib/jenkins_tests/vg.sh]: done

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1476/ for details (Mellanox internal link).

**Comment by rafiw** [comment]
@igor-ivanov due to hail10 i added support for cx4 (uses mlx4)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1477/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1481/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1482/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1483/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1485/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1490/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1491/ for details (Mellanox internal link).

**Comment by rafiw** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1493/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1495/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1499/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1501/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1505/ for details (Mellanox internal link).

File: contrib/jenkins_tests/vg.sh [hunk 4]
```diff
@@ -4,79 +4,120 @@ source $(dirname $0)/globals.sh
...
+			valgrind --log-file=${vg_dir}/${test_name}-valgrind-sr.log $vg_args \
+			$test_app_path sr ${test_opt} -i ${test_ip} > /dev/null 2>&1 &"
+		sleep 20
+		eval "LD_PRELOAD=$test_lib \
+			valgrind --log-file=${vg_dir}/${test_name}-valgrind-cl.log $vg_args \
+			$test_app_path pp ${test_opt} -i ${test_ip} -t 10"
+
+		if [ `ps -ef | grep $test_app | wc -l` -gt 1 ];
+		then
+			sudo pkill -SIGINT -f $test_app
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/vg.sh]: @rafiw please use `pkill -SIGINT -f $test_app 2>/dev/null || true` to suppress exit code form pkill

File: contrib/jenkins_tests/vg.sh [hunk 5]
```diff
@@ -4,79 +4,120 @@ source $(dirname $0)/globals.sh
...
+			sleep 10
+			# in case SIGINT didn't work
+			if [ `ps -ef | grep $test_app | wc -l` -gt 1 ];
+			then
+				sudo pkill -15 -f $test_app
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/vg.sh]: @rafiw please use `pkill -15 -f $test_app 2>/dev/null || true` to suppress exit code form pkill

File: contrib/jenkins_tests/vg.sh [hunk 6]
```diff
@@ -4,79 +4,120 @@ source $(dirname $0)/globals.sh
...
+				sleep 3
+			fi
+			if [ `ps -ef | grep $test_app | wc -l` -gt 1 ];
+			then
+				sudo pkill -9 -f $test_app
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/vg.sh]: @rafiw please use `pkill -SIGKILL -f $test_app 2>/dev/null || true` to suppress exit code form pkill

**Comment by igor-ivanov** [comment]
:+1:

**Comment by ilansmith** [comment]
Merged #424

**Pull request closed without merging**